imap/vcard_support.c

```diff
@@ -362,11 +363,28 @@ EXPORTED void vcard_to_v3(struct vparse_card *vcard)
     buf_free(&buf);
 }
 
+static int vcard_value_is_uri(const char *val)
```

Comment:
We want these changes on master even if we don't merge the HTTP GET changes, right?

Reply:
Wouldn't hurt to use this or a more sophisticated URI check in Cyrus. Maybe we also just want to merge that PR on the main branch, it's not that this is the first codepath where we rewrite stuff on GET anyway.